hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/NetgroupCache.java

```diff
@@ -55,6 +56,22 @@ public static void getNetgroups(final String user,
     }
   }
 
+  /**
```

Comment:
Why didn't you just change getNetGroups() to use Collection instead of List?

Reply:
Because I did not want to change the signature of the method as it could be seen as not recommended to change the public static method.
I decided to make a new method so later I can mark `getNetgroups` as `deprecated` and eventually remove it.

Reply:
I think in this case it makes more sense to just change the existing method.  A List is a Collection, so it should not break anything.  Simplifies the changes.